src/main/java/sda/training/rps/app/GameService.java

```diff
@@ -1,136 +1,317 @@
 package sda.training.rps.app;
 
 import sda.training.rps.dao.GameDao;
-import sda.training.rps.dao.IGameDao;
-import sda.training.rps.dao.IStageDao;
 import sda.training.rps.dao.StageDao;
 import sda.training.rps.model.Game;
 import sda.training.rps.model.Player;
 import sda.training.rps.util.Result;
 import sda.training.rps.util.ScannerSingleton;
 
-import java.util.Collections;
+import java.time.LocalDateTime;
+import java.time.ZoneOffset;
 import java.util.List;
+import java.util.Objects;
 
-public class GameService implements IGameService {
-    private IGameDao gameDao = new GameDao();
-    private IStageDao stageDao = new StageDao();
+public class GameService {
+    private final GameDao gameDao = new GameDao();
+    private final StageDao stageDao = new StageDao();
     private Game game = new Game();
-    private IStageService stageService = new StageService();
-    private ScannerSingleton scanner = ScannerSingleton.getInstance();
+    private final PlayerService playerService;
+    private final StageService stageService;
+    private final ScannerSingleton scanner = ScannerSingleton.getInstance();
+
+    public GameService(PlayerService playerService) {
+        this.playerService = playerService;
+        this.stageService = new StageService(playerService);
+    }
 
-    @Override
     public void startNewGame(Player player) {
         int winStagesNo = loadStagesProperty();
         game = createGame(player, winStagesNo);
         loopStages();
-
-        /*      wczytuje od gracz i ustawia ilosc wygranych etapow
-         *      Twozy instancje gry
-         *      petla stagy
-         * */
     }
 
     private int loadStagesProperty() {
-        return 0;
+        System.out.print("Podaj ile punktów potrzeba na wygraną: ");
+        return scanner.scannerInt();
     }
 
     private Game createGame(Player player, int winStagesNo) {
+        game.setId(null);
         game.setWinStagesNo(winStagesNo);
         game.setPlayer(player);
-        //ustawienie poczatku gry
+        game.setComputerScore(0);
+        game.setPlayerScore(0);
+        game.setStartDate(LocalDateTime.now());
+
         return gameDao.mergeObject(game);
     }
 
     private void loopStages() {
         Result result;
         boolean continueGame = true;
-        while (continuationGame(continueGame)) {
+        while (isContinuable(continueGame)) {
             result = stageService.playStage(game);
             addPoints(result);
-            continueGame = false;
+            System.out.println("Ty: " + game.getPlayerScore() + " \tPrzeciwnik: " + game.getComputerScore());
+            if (isContinuable(true)) {
```

Comment:
skoro zawsze przekazujesz tutaj true to jaki jest sens przekazywania tego parametru :D?